### PR TITLE
Clean up in CellIDTranslator

### PR DESCRIPTION
**Pull request by peterrum**
as pointed out by @drwells in https://github.com/dealii/dealii/pull/12513

### CODE REVIEW

File: include/deal.II/grid/cell_id_translator.h [hunk 1]
```diff
@@ -62,25 +62,26 @@ namespace internal
 
     /**
      * Return maximum number of cells, i.e., in the case the triangulation
-     * would be refined globally.
+     * ere globally refined `n_global_levels - 1` times.
```
Comment (drwells) [inline include/deal.II/grid/cell_id_translator.h:65]:
```suggestion
     * were globally refined `n_global_levels - 1` times.
```

**Review by drwells — CHANGES_REQUESTED** [review]
There is one small typo but otherwise 👍🏻

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/grid/cell_id_translator.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/grid/cell_id_translator.h]: Fixed!

**Review by drwells — APPROVED** [review]

**Comment by tjhei** [comment]
```
In file included from /jenkins/workspace/dealii_PR-12539/tests/all-headers/test_header.cc:17:0:

/jenkins/workspace/dealii_PR-12539/include/deal.II/grid/cell_id_translator.h:76:21: error: ‘TriaIterator’ does not name a type

The following tests FAILED:
	5036 - all-headers/grid/cell_id_translator.h.debug (Failed)
```

**Pull request merged**